### PR TITLE
Adding channels section, that does addition in parallel

### PR DESCRIPTION
**Pull request by mattnenterprise**

### CODE REVIEW

File: 04-channels/main.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+func Add(values []int, result chan<- int) {
+	result <- 0
+}
+
+// Write a function that partitions the slice into
+// partitions of size partitionSize
+// Ex:
+// 	values: [1, 2, 3, 4, 5, 6, 7]
+// 	partitionSize: 3
+//  result: [[1, 2, 3], [4, 5, 6], [7]]
```
Comment (kelcecil) [inline 04-channels/main.go:14]: I like your example input and output for functions. Fantastic idea!

File: 04-channels/solutions/main.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+//  result: [[1, 2, 3], [4, 5, 6], [7]]
+func PartitionSlice(values []int, partitionSize int) [][]int {
+	partitions := [][]int{}
+	length := len(values)
+	for i := 0; i < length; i += partitionSize {
+		end := i + partitionSize
+		if end >= length {
+			end = length
+		}
+		partitions = append(partitions, values[i:end])
```
Comment (kelcecil) [inline 04-channels/solutions/main.go:27]: Remind me that I'd better not forget to include slice indice selection in the slides.

File: 04-channels/solutions/main_test.go [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+}
+
+func GetAddTestData() AddTestData {
+	return AddTestData{
+		values:   []int{1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19},
+		expected: 190,
+	}
+}
+
+func BenchmarkAdd(b *testing.B) {
```
Comment (kelcecil) [inline 04-channels/solutions/main_test.go:81]: This will be a good section for teaching how to do benchmarks. Good idea!

Comment (mattnenterprise) [inline 04-channels/solutions/main_test.go:81]: I also thought it would show a way to prove parallel code is faster.

Comment (kelcecil) [inline 04-channels/solutions/main_test.go:81]: Very good thought! You'll definitely want to point this out to students!

**Pull request merged**